### PR TITLE
Improve CI script

### PR DESCRIPTION
**Pull request by masterleinad**
Apply a suggestion in  https://github.com/ORNL-CEES/mfmg/pull/159#discussion_r275197426 and run ci with `-Werror` so we don't forget to address warnings in pull requests.

### CODE REVIEW

File: ci/compile_and_run.sh [hunk 1]
```diff
@@ -13,10 +14,10 @@ ARGS=(
   -D DEAL_II_DIR=${DEAL_II_DIR}
   -D MFMG_ENABLE_AMGX=ON
   -D AMGX_DIR=${AMGX_DIR}
-  -D CMAKE_CXX_FLAGS="-Wall -Wpedantic -Wextra -Wshadow"
+  -D CMAKE_CXX_FLAGS="-Wall -Wpedantic -Wextra -Wshadow -Werror"
```
Comment (dalg24) [inline ci/compile_and_run.sh]: Is that really a good idea?

Comment (masterleinad) [reply ci/compile_and_run.sh]: What are your objections?

Comment (dalg24) [reply ci/compile_and_run.sh]: Won't even run the tests if you have a warning that you might have missed cause you use a different compiler

Comment (masterleinad) [reply ci/compile_and_run.sh]: For me, `CI` should check for all the requirements we want a pull request/the code to fulfill in the end and not so much as a development tool when creating a pull request.

In `deal.II`, we just have multiple checks were one would one would not have `-Werror` and run tests while another one would check, e.g. for indentation (not as a test) and for compiler warnings.

I would also be in favor to run `clang-tidy` if we want to enforce its checks in the end.

File: ci/compile_and_run.sh [hunk 2]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
-cd $1 || exit 1
+set -e
```
[the remark this answers is not in the document]
Comment (Rombur) [reply ci/compile_and_run.sh]: done

File: ci/compile_and_run.sh (hunk 1 above)
Comment (Rombur) [reply ci/compile_and_run.sh]: What do you guys want to do here? I don't have a strong opinion either way

File: ci/compile_and_run.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (masterleinad) [reply ci/compile_and_run.sh]: I am using this script for testing pull requests locally and would like to keep this option (or go with the previous version).

**Pull request merged**